src/util/sss_ptr_hash.c

```diff
@@ -144,22 +114,42 @@ sss_ptr_hash_delete_cb(hash_entry_t *item,
         return;
     }
 
-    callback_entry.key = item->key;
-    callback_entry.value.type = HASH_VALUE_PTR;
-    callback_entry.value.ptr = value->ptr;
-
-    /* Free value, this also will disable spy */
-    talloc_free(value);
-
     if (pvt != NULL) {
         /* Switch to the input value and call custom callback. */
         data = talloc_get_type(pvt, struct sss_ptr_hash_delete_data);
         if (data == NULL) {
             DEBUG(SSSDBG_CRIT_FAILURE, "Invalid data!\n");
-            return;
+        } else {
+            callback_entry.key = item->key;
+            callback_entry.value.type = HASH_VALUE_PTR;
+            callback_entry.value.ptr = value->payload;
+            /* Even if execution is already in the context of
+             * talloc_free(payload) -> talloc_free(value) -> ...
+             * (i.e. value->being_destroyed) there are still might be
```

Comment:
"there are still might be" scratch "are"

Reply:
Fixed.